tips/TIP-0052/tip-0052.md

```diff
@@ -0,0 +1,345 @@
+    <td>
+      <b>Description</b>
+    </td>
+  </tr>
+  <tr>
+    <td>Unlock Type</td>
+    <td>uint8</td>
+    <td>Set to <strong>value 5</strong> to denote a <i>Multi Unlock</i>.</td>
+  </tr>
+  <tr>
+    <td>Unlocks Count</td>
+    <td>uint8</td>
+    <td>The number of unlocks following.</td>
+  </tr>
+  <tr>
+    <td valign="top">Unlocks <code>anyOf</code></td>
+    <td colspan="2">
+      <details>
+        <summary>Signature Unlock</summary>
+        <blockquote>Unlocks the address derived from the contained Public Key in the transaction in which it is contained in. Defined in <a href='../TIP-0045/tip-0045.md#signature-unlock'>TIP-45 (Signature Unlock)</a>.</blockquote>
+      </details>
+      <details>
+        <summary>Reference Unlock</summary>
+        <blockquote>References a previous unlock to support unlocking multiple inputs owned by the same address. Defined in <a href='../TIP-0045/tip-0045.md#reference-unlock'>TIP-45 (Reference Unlock)</a>.</blockquote>
+      </details>
+      <details>
+        <summary>Account Unlock</summary>
+        <blockquote>Points to the unlock of a consumed Account Output. Defined in <a href='../TIP-0042/tip-0042.md#account-unlock'>TIP-42 (Account Unlock)</a>.</blockquote>
+      </details>
+      <details>
+        <summary>Anchor Unlock</summary>
+        <blockquote>Points to the unlock of a consumed Anchor Output. Defined in <a href='../TIP-0054/tip-0054.md#anchor-unlock'>TIP-54 (Anchor Unlock)</a>.</blockquote>
+      </details>
+      <details>
+        <summary>NFT Unlock</summary>
+        <blockquote>Points to the unlock of a consumed NFT Output. Defined in <a href='../TIP-0043/tip-0043.md#nft-unlock'>TIP-43 (NFT Unlock)</a>.</blockquote>
+      </details>
+      <details>
+        <summary>Empty Unlock</summary>
+        <blockquote>Used to maintain correct index relationship between addresses and signatures when unlocking a Multi Address where not all addresses are unlocked. Defined in <a href='../TIP-0052/tip-0052.md#empty-unlock'>TIP-52 (Empty Unlock)</a>.</blockquote>
+      </details>
+    </td>
+  </tr>
+</table>
+
+#### Additional syntactic transaction validation rules
+
+A Multi Unlock is only valid if all of the following conditions hold:
+
+- `1 <= Unlocks Count <= 10`.
+
+#### Additional semantic transaction validation rules
+
+A transaction containing a Multi Unlock is only valid if all of the following conditions hold:
+
+- The corresponding Address of the to-be-unlocked input is of type _Multi Address_.
+- The `Addresses Count` of the Multi Address and the `Unlocks Count` of the Multi Unlock must match.
+- If `Unlocks` contains Reference, Account or NFT Unlocks, they must point to an Unlock outside of the Multi Unlock.
+- Let the `Cumulative Unlocked Weight` be the sum of weights of the addresses that were successfully unlocked, through
+  Unlocks other than _Empty Unlocks_, which do not add to this sum.
+- `Cumulative Unlocked Weight >= Threshold`.
+
+### Empty Unlock
```

Comment:
Should it be a syntactic rule that it can't be used outside of a MultiUnlock or are we fine waiting semantic and see that address/unlock is not a good match?

Reply:
The schemas tell you where the Empty Unlock is allowed - only in Multi Unlocks.
Same for implicit account creation addresses. They are also not allowed in Address Unlock Conditions within Accounts, but are allowed in Basic Outputs, neither are they allowed in Expiration Unlock Conditions. Those rules are not defined anywhere, the schema itself is the rule.

Since we already have the schemas and they are much easier to maintain in the TIPs I would like to avoid duplicating those rules in text were it cannot be automatically updated.